src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -41,6 +44,53 @@ class DbTables(int, Enum):
     # ifOutQLen ::= { ifEntry 21 }
     SAI_PORT_STAT_IF_OUT_QLEN = 21
 
+class ArpUpdater(MIBUpdater):
+    def __init__(self):
+        super().__init__()
+        self.arp_dest_map = {}
+        self.arp_dest_list = []
+        # call our update method once to "seed" data before the "Agent" starts accepting requests.
+        self.update_data()
+
+    def update_data(self):
+        self.arp_dest_map = {}
+        self.arp_dest_list = []
+        for entry in python_arptable.get_arp_table():
+            dev = entry['Device']
+            mac = entry['HW address']
+            ip = entry['IP address']
+
+            if_index = mibs.get_index_from_str(dev)
+            if if_index is None: continue
+
+            mactuple = mac_decimals(mac)
+            machex = ''.join(chr(b) for b in mactuple)
+            # if MAC is all zero
+            #if not any(mac): continue
+
+            iptuple = ip2tuple_v4(ip)
+
+            subid = (if_index,) + iptuple
+            self.arp_dest_map[subid] = machex
+            self.arp_dest_list.append(subid)
+
+            # print(subid, dev, mac, ip)
```

Comment:
Removed

---
In reply to: [108499446](https://github.com/Azure/sonic-snmpagent/pull/19#discussion_r108499446) [](ancestors = 108499446)